Userland/Libraries/LibWeb/HTML/Policy.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, Noah Bright <noah.bright.1@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashTable.h>
+#include <AK/String.h>
+#include <LibWeb/HTML/Origin.h>
+
+// https://w3c.github.io/webappsec-csp/#framework-policy
+namespace Web::HTML {
+
+// https://w3c.github.io/webappsec-csp/#directives
+struct Directive {
+    String key;
+    HashTable<String> value;
+};
+// FIXME: implement directive algorithms
+
+class Policy {
+public:
+    auto& directive_set() { return m_directive_set; }
+
+private:
+    // an ordered set
+    Vector<Directive> m_directive_set;
```

Comment:
This should just be an `OrderedHashMap<String, Optional<HashTable<String>>>`

> Each [policy](https://w3c.github.io/webappsec-csp/#content-security-policy-object) contains an [ordered set]>?> (https://infra.spec.whatwg.org/#ordered-set) of directives (its [directive set](https://w3c.github.io/webappsec-csp/#policy-directive-set)), each of which controls a specific behavior. [...]

Ok, so we have a `OrderedHashMap< ' directive '>`


> Each [directive](https://w3c.github.io/webappsec-csp/#directives) is a name / value pair. The [name](https://w3c.github.io/webappsec-csp/#directive-name) is a non-empty [string](https://infra.spec.whatwg.org/#string), and the [value](https://w3c.github.io/webappsec-csp/#directive-value) is a [set](https://infra.spec.whatwg.org/#ordered-set) of non-empty [strings](https://infra.spec.whatwg.org/#string).

Ok, `OrderedHashMap<String, HashTable<String>>`

> The [value](https://w3c.github.io/webappsec-csp/#directive-value) MAY be [empty](https://infra.spec.whatwg.org/#list-is-empty).

So it's actually a `OrderedHashMap<String, Optional<HashTable<String>>>`